### DISCUSSION TITLE
Can I have a permanent window that shows only results of last run?

### DISCUSSION THREAD

**Discussion by WhoIsSethDaniel** [nvim-neotest/neotest#277] [Q&A]
I have one question: 

Can I have a 'permanent' window that only shows the results of the most recent run? I'd especially like it if I can watch the output as it happens. I have tried setting `neotest.output.open.open_win` to a function. When I run a test (or tests) I can then open the window using `neotest.output.open({ open_win = function(...), last_run = true, ...`. This works, but the next run of the test(s) doesn't refresh the window. I have to enter the window, close it, and re-open it. I tried adding the `neotest.output.open(...)` directly after the `neotest.run.run()` but the window continued to show the previous run. I think what I want is something like `output_panel` but that clears the window before each run. The output is useful for me except that it has lots of other runs...none of which I care about. It can make it difficult to find what I am looking for. I tried clearing the `output_panel` just prior to `neotest.run.run()` but the output to the panel became garbled for reasons I am unclear about. I think maybe using the `output_panel` would be my preferred way, but I haven't been able to figure out why the output becomes so garbled.

I am using the `neotest-go` runner only, at this time.

**Comment by WhoIsSethDaniel** [comment]
I am doing this to clear the panel prior to run:
```lua
  return function()
    local buf = vim.fn.bufnr 'Neotest Output Panel'
    if buf ~= -1 then
      wb.set_options({ modifiable = true }, { buf = buf })
      vim.api.nvim_buf_set_lines(buf, 0, -1, false, {})
      wb.set_options({ modifiable = false }, { buf = buf })
    end
    neotest.run.run(opts)
  end
```
(in the above, `wb.set_options` is just a call to vim.api.nvim_buf_set_option)
It now seems less that the output isn't garbled but that the entire buffer is being re-written on each run. So this includes the earlier runs.

**Comment by WhoIsSethDaniel** [comment]
In looking at the `neotest` code for `output_panel` it seems to be reading from a file, presumably one that is written by the runner. It seems to read the entire file and sends it to a channel...which I guess is placed as the contents in the window. If that's correct my approach definitely won't work. Perhaps `output_panel` could have a method to truncate the file? Or an option to only show the most recent run (probably implemented by truncating the file prior to a run).

**Comment by quolpr** [comment]
@WhoIsSethDaniel did you find the solution? I have the same problem 🙂

**Comment by WhoIsSethDaniel** [comment]
No. `neotest` now includes [an api call](https://github.com/nvim-neotest/neotest/blob/6f35d797882c6ce0ab7c87dc86561512dc3d7aba/lua/neotest/consumers/output_panel/init.lua#L90-L94) for clearing the `output_panel` but it is doing it the same way I was doing it in the comment above. I don't know if the problem I mentioned (also above) is specific to the `neotest-go` adapter or not. I need to try the `neotest-golang` adapter to see if the behavior is more what I would expect. But since I currently only use the `neotest-go` adapter it is my only point of reference.
